prow/hook/metrics.go

```diff
@@ -30,23 +30,39 @@ var (
 		Name: "prow_webhook_response_codes",
 		Help: "A counter of the different responses hook has responded to webhooks with.",
 	}, []string{"response_code"})
+	configMapSizeGauges = prometheus.NewGaugeVec(prometheus.GaugeOpts{
+		Name: "prow_configmap_size_bytes",
+		Help: "Size of data fields in ConfigMaps updated automatically by Prow in bytes.",
+	}, []string{"name", "namespace"})
 )
 
 func init() {
 	prometheus.MustRegister(webhookCounter)
 	prometheus.MustRegister(responseCounter)
+	prometheus.MustRegister(configMapSizeGauges)
 }
 
 // Metrics is a set of metrics gathered by hook.
 type Metrics struct {
 	WebhookCounter  *prometheus.CounterVec
 	ResponseCounter *prometheus.CounterVec
+	*PluginMetrics
+}
+
+// PluginMetrics is a set of metrics that are gathered by plugins.
+// It is up the the consumers of these metrics to ensure that they
+// update the values in a thread-safe manner.
+type PluginMetrics struct {
+	ConfigMapGauges *prometheus.GaugeVec
 }
 
 // NewMetrics creates a new set of metrics for the hook server.
 func NewMetrics() *Metrics {
 	return &Metrics{
 		WebhookCounter:  webhookCounter,
 		ResponseCounter: responseCounter,
+		PluginMetrics: &PluginMetrics{
+			ConfigMapGauges: configMapSizeGauges,
```

Comment:
NIP: `configMapSizeGauge` as singular?

Reply:
Technically it's a slice of gauges -- one per map

Reply:
I see.